src/arch/arm/tlb.cc

```diff
@@ -58,8 +58,38 @@ namespace gem5
 
 using namespace ArmISA;
 
+TlbEntry*
+TLB::Table::accessEntry(const KeyType &key)
```

Comment:
I am going to dump my thoughts, but I am unfamiliar with the code, so there may be no action point:

I think we should override the other access function instead. It would be safer, as it guarantees that no one bypasses this functional check. That said, it seems that there is no guarantee that key.functional == entry.key.functional, right?

Reply:
I understand your rationale. However entry.key.functional does not mean anything. The functional boolean is a property of the access, not of the entry. It means: this is not a real workload lookup/access; I just want to get the entry to be able to do a functional VA/PA translation. Therefore

1) Do not update stats
2) Do not affect replacement policy/data

Another solution which lets us avoid overriding access Entry would be to not use accessEntry for functional accesses and handle things differently in the TLB::lookup method